### PR TITLE
Bump flyteidl version

### PR DESCRIPTION
**Pull request by pingsutw**
Signed-off-by: Kevin Su <pingsutw@apache.org>

# TL;DR
Bump flyteidl version to support Ray tasks. https://github.com/flyteorg/flyteidl/pull/308

## Type
 - [ ] Bug Fix
 - [ ] Feature
 - [x] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/2641

## Follow-up issue
_NA_

### CODE REVIEW

File: pkg/server/service.go [hunk 1]
```diff
@@ -401,6 +401,7 @@ func serveGatewaySecure(ctx context.Context, pluginRegistry *plugins.Registry, c
 			Certificates: []tls.Certificate{*cert},
 			NextProtos:   []string{"h2"},
 		},
+		ReadHeaderTimeout: 0,
```
Comment (katrogan) [inline pkg/server/service.go:404]: why 0?

Comment (katrogan) [reply pkg/server/service.go:404]: shouldn't we specify a non-zero value then to address the actual slowloris vulnerability?

Comment (pingsutw) [reply pkg/server/service.go]: yes, added it to the server config and used 32s as the default value which is same as the [k8s API server](https://github.com/kubernetes/kubernetes/blob/75b19b242c34f7c1d1545e156f802f0a06b90533/staging/src/k8s.io/apiserver/pkg/server/secure_serving.go#L172).

File: boilerplate/flyte/end2end/run-tests.py [hunk 2]
```diff
@@ -98,10 +98,15 @@ def executions_finished(executions_by_wfgroup: Dict[str, List[FlyteWorkflowExecu
...
-        for execution in executions:
-            print(f"About to sync execution_id={execution.id.name}")
-            remote.sync(execution)
+    try:
+        for executions in executions_by_wfgroup.values():
+            for execution in executions:
+                print(f"About to sync execution_id={execution.id.name}")
+                remote.sync(execution)
+    except:
+        print("GOT TO THE EXCEPT")
```
Comment (katrogan) [inline boilerplate/flyte/end2end/run-tests.py:107]: remove this?

File: pkg/config/config.go [hunk 3]
```diff
@@ -95,6 +96,7 @@ var defaultServerConfig = &ServerConfig{
 			MaxExpiresIn: config.Duration{Duration: time.Hour},
 		},
 	},
+	ReadHeaderTimeoutSeconds: 32,
```
Comment (katrogan) [inline pkg/config/config.go:99]: mind adding a similar comment as kubernetes does for why this value?

File: boilerplate/flyte/end2end/run-tests.py (hunk 2 above)
Comment (pingsutw) [reply boilerplate/flyte/end2end/run-tests.py]: This is auto-generated by the script. I will update the file in flyteorg/boilerplate as well. https://github.com/flyteorg/boilerplate/blob/master/boilerplate/flyte/end2end/run-tests.py#L100-L108

File: pkg/config/config.go (hunk 3 above)
Comment (pingsutw) [reply pkg/config/config.go]: updated it

File: boilerplate/flyte/end2end/run-tests.py (hunk 2 above)
Comment (katrogan) [reply boilerplate/flyte/end2end/run-tests.py]: oh I see! was this change in boilerplate intentional?

File: boilerplate/flyte/end2end/run-tests.py [hunk 4]
```diff
@@ -98,10 +98,14 @@ def executions_finished(executions_by_wfgroup: Dict[str, List[FlyteWorkflowExecu
...
-        for execution in executions:
-            print(f"About to sync execution_id={execution.id.name}")
-            remote.sync(execution)
+    try:
+        for executions in executions_by_wfgroup.values():
+            for execution in executions:
+                print(f"About to sync execution_id={execution.id.name}")
+                remote.sync(execution)
+    except:
+        print("COUNT THIS!")
```
Comment (katrogan) [inline boilerplate/flyte/end2end/run-tests.py:107]: is this intentional too?

**Review by katrogan — APPROVED** [review]

**Pull request merged**